src/celeritas/decay/interactor/MuDecayInteractor.hh

```diff
@@ -0,0 +1,299 @@
+            if (generate_canonical(rng)
+                <= electron_nu_energy_frac
+                       * (real_type{1} - electron_nu_energy_frac))
+            {
+                break;
+            }
+            electron_nu_energy_frac = sample_max_;
+        }
+
+        electron_energy_frac = generate_canonical(rng);
+        if (electron_nu_energy_frac >= real_type{1} - electron_energy_frac)
+        {
+            break;
+        }
+        electron_nu_energy_frac = real_type{1} - electron_energy_frac;
+    }
+    real_type muon_nu_energy_frac = real_type{2} - electron_energy_frac
+                                    - electron_nu_energy_frac;
+
+    // Momentum of secondaries at rest frame; neutrino masses are ignored
+    auto charged_lep_energy
+        = this->momentum(electron_energy_frac, shared_.electron_mass);
+    auto electron_nu_energy = this->momentum(electron_nu_energy_frac, 0);
+    auto muon_nu_energy = this->momentum(muon_nu_energy_frac, 0);
+
+    // Angle between charged lepton and electron neutrino
+    real_type cos_theta
+        = real_type{1} - 2 / electron_energy_frac - 2 / electron_nu_energy_frac
+          + 2 / (electron_nu_energy_frac * electron_energy_frac);
+    CELER_ASSERT(std::fabs(cos_theta) <= 1);
+    real_type sin_theta = std::sqrt(real_type{1} - ipow<2>(cos_theta));
+
+    // Use an initial arbitrary direction for secondaries.
+    // Placing the electron on one axis (e.g. +z), simplifies the calculation
+    // for the other two secondaries. The direction of the electron neutrino
+    // is defined by the sampled theta, and the muon neutrino direction is
+    // calculated via conservation of momentum
+    Real3 charged_lep_dir = {0, 0, 1};
+    Real3 electron_nu_dir = {sin_theta, 0, cos_theta};
+    Real3 muon_nu_dir
+        = {-(electron_nu_energy_frac / muon_nu_energy_frac) * sin_theta,
+           0,
+           -electron_energy_frac / muon_nu_energy_frac
+               - (electron_nu_energy_frac / muon_nu_energy_frac) * cos_theta};
+
+    // Apply a spherically uniform rotation to the decay
+    real_type euler_phi
+        = UniformRealDistribution<real_type>(0, 2 * constants::pi)(rng);
+    real_type euler_theta = std::acos(2 * generate_canonical(rng) - 1);
+    real_type euler_psi
+        = UniformRealDistribution<real_type>(0, 2 * constants::pi)(rng);
+
+    EulerRotation rotate(euler_phi, euler_theta, euler_psi);
+    charged_lep_dir = rotate(charged_lep_dir);
+    electron_nu_dir = rotate(electron_nu_dir);
+    muon_nu_dir = rotate(muon_nu_dir);
+
+    // Boost secondaries to the lab frame
+    auto charged_lep_4vec = this->to_lab_frame(
+        charged_lep_dir, charged_lep_energy, shared_.electron_mass);
+    auto electron_nu_4vec
+        = this->to_lab_frame(electron_nu_dir, electron_nu_energy, 0);
+    auto muon_nu_4vec = this->to_lab_frame(muon_nu_dir, muon_nu_energy, 0);
+
+    // \todo Return electron only?
```

Comment:
Looking great @stognini! Still have to go over most of this, but since we were talking about this point on the call, I guess my preference would be to just create the electron secondary until we have a use case for the neutrinos. The extra bit of computation won't be a big deal, but the two extra secondaries per decay (that we can't track anyway) could increase the secondary/initializer memory requirements a bit. And since you've already implemented it and we'll have it in the history, we can easily just revert the change when we need it. (I'll let @sethrj and @whokion weigh in, though.)

Reply:
I do agree with @amandalund for now (unless we want to add neutrino-nucleus processes in a near-term future or want to send these neutrinos back to CPU so that Geant4 can handle them in the case that users add neutrino processes in their physics list, of which data transfer needs to be supported in a similar way when we add  lepto-/photo-nuclear processes).  Of course, please share any other users cases that you are considering. 

Regarding to the maximum energy of the daughter electron from the muon decay, I think that the formulas in the P.M. is an approximation (i.e, assuming that secondaries are massless in the V-A calculation except the primary muon).  The implementation in Geant4 may be more practically/experically correct.  You may still recover the energy conservation in the rest frame of the muon following correct kinematics between (e + n_e) + n_muon and muon as necessary - I am surprised that Geant4 does not calculate them correctly.  Anyway, I will also review and give more detailed comments later.  Thanks for a good addition!